src/JSInterop/Microsoft.JSInterop/src/JSRuntimeBase.cs

```diff
@@ -94,10 +94,17 @@ internal void EndInvokeJS(long asyncHandle, bool succeeded, JSAsyncCallResult as
                 if (succeeded)
                 {
                     var resultType = TaskGenericsUtil.GetTaskCompletionSourceResultType(tcs);
-                    var result = asyncCallResult != null ?
-                        JsonSerializer.Parse(asyncCallResult.JsonElement.GetRawText(), resultType, JsonSerializerOptionsProvider.Options) :
-                        null;
-                    TaskGenericsUtil.SetTaskCompletionSourceResult(tcs, result);
+                    try
+                    {
+                        var result = asyncCallResult != null ?
+                            JsonSerializer.Parse(asyncCallResult.JsonElement.GetRawText(), resultType, JsonSerializerOptionsProvider.Options) :
+                            null;
+                        TaskGenericsUtil.SetTaskCompletionSourceResult(tcs, result);
+                    }
+                    catch (Exception exception)
+                    {
+                        TaskGenericsUtil.SetTaskCompletionSourceException(tcs, new JSException(exception.Message, exception));
```

Comment:
Should this have a different message since we're wrapping it? Like `An exception occurred executing JS interop. See InnerException for details.`